core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/SubmissionApiServiceSpec.scala

```diff
@@ -662,4 +668,60 @@ class SubmissionApiServiceSpec extends ApiServiceSpec {
       )
     }
   }
+
+  it should "return 200 when reading a Google Genomics operation with PAPIv1 job id" in withEmptyTestDataApiServices { services =>
+    withStatsD {
+      Get(s"/workflows/workflow_with_job_ids/genomics/operations/dummy-job-id") ~> services.sealedInstrumentedRoutes ~>
+        check {
+          assertResult(StatusCodes.OK, responseAs[String]) {
+            status
+          }
+          // message returned by MockGoogleServicesDAO
+          assertResult("""{"foo":"bar"}""".parseJson.asJsObject) {
+            responseAs[JsObject]
+          }
+        }
+    } { capturedMetrics =>
+      val wsPathForRequestMetrics = "workflows.redacted.genomics.redacted.redacted"
+      val expected = expectedHttpRequestMetrics("get", wsPathForRequestMetrics, StatusCodes.OK.intValue, 1)
+      assertSubsetOf(expected, capturedMetrics)
+    }
+  }
+
+  it should "return 200 when reading a Google Genomics operation with PAPIv2 job id" in withEmptyTestDataApiServices { services =>
+    withStatsD {
+      Get(s"/workflows/workflow_with_job_ids/genomics/projects/dummy-project/operations/dummy-job-id") ~> services.sealedInstrumentedRoutes ~>
+        check {
+          assertResult(StatusCodes.OK) {
+            status
+          }
+          // message returned by MockGoogleServicesDAO
+          assertResult("""{"foo":"bar"}""".parseJson.asJsObject) {
+            responseAs[JsObject]
+          }
+        }
+    } { capturedMetrics =>
+      val wsPathForRequestMetrics = "workflows.redacted.genomics.redacted.redacted.redacted.redacted"
+      val expected = expectedHttpRequestMetrics("get", wsPathForRequestMetrics, StatusCodes.OK.intValue, 1)
+      assertSubsetOf(expected, capturedMetrics)
+    }
+  }
+
+  it should "return 404 when reading a Google Genomics operation for a non-existent workflow" in withEmptyTestDataApiServices { services =>
+    Get(s"/workflows/bogus/genomics/projects/dummy-project/operations/dummy-job-id") ~> services.submissionRoutes ~>
+      check {
+        assertResult(StatusCodes.NotFound) {
+          status
+        }
+      }
+  }
+
+  it should "return 404 when reading a Google Genomics operation for a non-existent job" in withEmptyTestDataApiServices { services =>
+    Get(s"/workflows/workflow_with_job_ids/genomics/projects/dummy-project/operations/bogus") ~> services.submissionRoutes ~>
+      check {
+        assertResult(StatusCodes.NotFound) {
+          status
+        }
+      }
+  }
```

Comment:
Should there be a test for the case where the job exists but CromIAM denies you permission?

Reply:
that is up to cromiam I think

Reply:
more specifically, from outside cromiam, a job that does not exist looks the same as a job where the user does not have access. There is a test where our mock cromiam returns 404  so the rawls tests cover all the cases expected from cromiam.